lib/wal/wal.c

```diff
@@ -719,15 +725,19 @@ wal_close(struct wal *wal)
         mutex_unlock(&wal->sync_mutex);
 
         pthread_join(wal->sync_notify_tid, 0);
-        cv_destroy(&wal->sync_cv);
-        mutex_destroy(&wal->sync_mutex);
     }
 
     /* Write a close record to indicate graceful shutdown */
     if (!wal->read_only)
         wal_mdc_close_write(wal->mdc);
     wal_mdc_close(wal->mdc);
 
+    mutex_destroy(&wal->sync_mutex);
```

Comment:
Greg, is this the fix for the sync_mutex abort issue that you ran into during close? Was this mutex destroyed much earlier than it should have?

Reply:
I think at the time I had some other issues with my workqueue changes, so I don't think you'll see any problems now.  Pretty sure I ran 1,8 for a long time before I reverted it.  But would be good for you to confirm!

Reply:
BTW, here's how I was building at that time:

$ meson setup --buildtype=debug -Dc_args='-Og -g3' -Drpath=true ...

Reply:
Thanks!